pkg/block/adapter.go

```diff
@@ -10,8 +10,10 @@ import (
 
 // MultipartPart single multipart information
 type MultipartPart struct {
-	ETag       string
-	PartNumber int
+	ETag         string
+	PartNumber   int
+	LastModified time.Time
+	Size         int64
```

Comment:
For this PR adding S3 support it doesn't matter, but I think it _will_ matter when we add support on Azure: an ongoing MPU will _not_ have these values.  So I would prefer that the new fields be optional, and then be handled by the caller.

As an alternative, you could add an error and document that the ListParts interface returns it for an MPU that started before ListParts support was added.

Reply:
These already exist in the serde struct (and cannot be optional) they will just get the default zero value